app/src/main/java/com/example/poste/activities/Shared_Folder_v2.java

```diff
@@ -45,34 +45,42 @@ protected void onCreate(Bundle savedInstanceState) {
         getSupportActionBar().hide();
 
         // Set the activity layout
-        setContentView(R.layout.activity_shared_folder);
+        setContentView(R.layout.activity_shared_folder_v2);
 
-        TextView folderNameView = findViewById(R.id.share_folder_name);
-        EditText emailView = findViewById(R.id.share_folder_email);
+        TextView folderNameView = findViewById(R.id.textViewShareFolderName);
+        EditText emailView = findViewById(R.id.editTextEmailToShareWith);
+        TextView folderPermissionsView = findViewById(R.id.textViewShareFolderPermissions);
 
-        Spinner spinner = findViewById(R.id.share_folder_spinner);
-        ArrayAdapter<CharSequence> spinnerAdapter = ArrayAdapter.createFromResource(this, R.array.spinner_values, android.R.layout.simple_spinner_item);
-        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        spinner.setAdapter(spinnerAdapter);
+        Spinner folderSpinner = findViewById(R.id.share_folder_spinner);
+        Spinner permissionsSpinner = findViewById(R.id.share_folder_permissions_spinner);
+        ArrayAdapter<CharSequence> folderSpinnerAdapter = ArrayAdapter.createFromResource(this, R.array.spinner_values, android.R.layout.simple_spinner_item);
+        folderSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        folderSpinner.setAdapter(folderSpinnerAdapter);
+        ArrayAdapter<CharSequence> permissionsSpinnerAdapter = ArrayAdapter.createFromResource(this, R.array.spinner_values, android.R.layout.simple_spinner_item);
+        permissionsSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        permissionsSpinner.setAdapter(permissionsSpinnerAdapter);
 
         Intent intent = getIntent();
         folderNameView.setText(intent.getStringExtra("folderName"));
         int folderId = intent.getIntExtra("folderId", -1);
         String folderName = intent.getStringExtra("folderName");
 
-        Button saveBtn = findViewById(R.id.share_folder_save_btn);
-        Button cancelBtn = findViewById(R.id.share_folder_cancel_btn);
+        Button saveBtn = findViewById(R.id.buttonSaveChanges);
+        Button cancelBtn = findViewById(R.id.buttonCancelChanges);
 
         saveBtn.setOnClickListener(view -> {
-            Intent newIntent = new Intent(Shared_Folder.this, DashboardActivity.class);
-            String selectedAccessValue = spinner.getSelectedItem().toString();
+            Intent newIntent = new Intent(Shared_Folder_v2.this, DashboardActivity.class);
+            String folderSelectedAccessValue = folderSpinner.getSelectedItem().toString();
+            String permissionsSelectedAccessValue = permissionsSpinner.getSelectedItem().toString();
             try {
                 Folder targetFolder = API.getFolderById(folderId);
```

Comment:
These API calls have been deprecated in favor of the new HTTP helper methods / retrofit client; the other activities will have examples. See com.example.poste.http package for more context